### PR TITLE
[WIP] Added very basic elasticsearch reporter recipe

### PR DESCRIPTION
**Pull request by mablae**
This is a very simple task to report an deployment in to 
an elasticsearch cluster. 

This is used by me to display the deployments as annotations in grafana or kibana.

### CODE REVIEW

File: recipes/elasticsearch.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    $basicQueryString = sprintf('curl %s -XPOST %s://%s/%s/%s/  -d "%s"',
+        env('elasticsearch_basic_auth'),
+        $config['protocol'],
+        $config['host'],
+        $config['host'],
+        env(elasticsearch_index_name()),
+        $config['doc_type'],
+        json_encode($payload)
+        );
```
Comment (oanhnn) [inline recipes/elasticsearch.php]: Can you fix coding style.
```
$basicQueryString = sprintf('curl %s -XPOST %s://%s/%s/%s/  -d "%s"',
    env('elasticsearch_basic_auth'),
    $config['protocol'],
    $config['host'],
    $config['host'],
    env(elasticsearch_index_name()),
    $config['doc_type'],
     json_encode($payload)
);
```

**Comment by oanhnn** [comment]
Can you add document for this recipe

**Comment by mablae** [comment]
@oanhnn Sure, will do... Also rebase and such squashing. Look at the mess I produced :smile: 

I finally got it working at least. Feedback welcome

File: recipes/elasticsearch.php (hunk 1 above)
Comment (mablae) [inline recipes/elasticsearch.php]: What would be best to fix it? Is deployer following a specific CS ? Or do just talk about that space in front of `json_encode`?

File: recipes/elasticsearch.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    'username' => null,
+    'password' => null,
+    'insecure' => false,
+]);
+
+env('elasticsearch_basic_auth', function() {
+    $config = get('elasticsearch');
+
+    $basicAuth = '';
+    if ($config['username'] && $config['password'] && $config['protocol'] === 'https') {
```
Comment (oanhnn) [inline recipes/elasticsearch.php]:
```
$php -r "if('0') echo 'TRUE'; else echo 'FALSE';"
FALSE
```

Comment (mablae) [inline recipes/elasticsearch.php]: I will add more input sanitizing, thank you for you feedback.

**Comment by elfet** [comment]
Close as work not finished.

**Pull request closed without merging**